### PR TITLE
Update normal geometries

### PR DESCRIPTION
**Pull request by loumalouomega**
#915

### CODE REVIEW

**Comment by philbucher** [comment]
I discussed it at the chair with @KlausBSautter and what we propose is to specify an additional vector for computing the normal
Either this vector is perpenticular to the normal that is being computed (like the Z-Axis in 2D) or we specify the a vector in the plane where we want to compute the normal (any vector in the X-Y Plane in 2D)

**Comment by loumalouomega** [comment]
Well, at least this solves the problem of checking the non-zero normal

File: kratos/geometries/line_3d_2.h [hunk 1]
```diff
@@ -671,12 +671,6 @@ class Line3D2 : public Geometry<TPointType>
...
         array_1d<double,3> normal = ZeroVector(3);
-        
-        // We get the local points
-        const TPointType& first_point  = BaseType::GetPoint(0);
-        const TPointType& second_point = BaseType::GetPoint(1);
- 
-        // TODO: FINISH ME
  
         // We normalize
         const double norm_normal = norm_2(normal);
```
Comment (RiccardoRossi) [inline kratos/geometries/line_3d_2.h]: you should throw an error if the normal's norm is zero. AT LEAST IN DEBUG

it means that the geometry is degradated

**Review by RiccardoRossi — COMMENTED** [review]
@philbucher, that additional vector should be the LOCAL_AXES_2, which the GiD interface is already capable of providing

**Comment by RiccardoRossi** [comment]
correcting what i said above, the geometry cannot use variables.

i added the use of the extra normal in the line_load_condition

File: kratos/geometries/geometry.h [hunk 2]
```diff
@@ -727,7 +727,9 @@ class Geometry : public PointerVector<TPointType>
 
         array_1d<double, 3> normal;
         MathUtils<double>::CrossProduct(normal, tangent_xi, tangent_eta);
-        return normal/norm_2(normal);
+	const double norm_normal = norm_2(normal);
+	if (norm_normal > 0.0) normal /= norm_normal;
+        return normal;
```
Comment (RiccardoRossi) [inline kratos/geometries/geometry.h]: you should throw an error if the norm_normal < eps

Comment (loumalouomega) [reply kratos/geometries/geometry.h]: In debug or always?

**Comment by loumalouomega** [comment]
@RiccardoRossi done

**Review by RiccardoRossi — APPROVED** [review]

**Comment by pooyan-dadvand** [comment]
I strongly disagree with throwing error in normal method. I have cases where having a zero normal would not harm unless you throw and error....

**Comment by RiccardoRossi** [comment]
Well,
        then the method is badly named. I think we should have Normal() (giving the area-proportional normal and eventually giving back a zero, since it is not made unit) and UnitNormal() which gives back a unit vector and hence should throw an error if this is impossible.

The problem is how to change this, making it clear

**Comment by pooyan-dadvand** [comment]
I like this solution. And actually the area proportional normal is more util for me than the unit one...

**Comment by loumalouomega** [comment]
Do I change it?

**Comment by pooyan-dadvand** [comment]
I think @RiccardoRossi is on it.

**Comment by loumalouomega** [comment]
Ok, but include me, I use the normal a lot, and basically I use the unit normal

**Comment by RiccardoRossi** [comment]
yes, i am on it, i opened a new branch for doing it

**Pull request merged**